assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -8425,6 +8425,66 @@ else if (groupChoice != null)
 				commitAssignmentEdit(state, post, ac, a, title, visibleTime, openTime, dueTime, closeTime, enableCloseDate, section, range, groups, isGroupSubmit, 
 						usePeerAssessment,peerPeriodTime, peerAssessmentAnonEval, peerAssessmentStudentViewReviews, peerAssessmentNumReviews, peerAssessmentInstructions);
 
+                // Locking and unlocking groups   
+                List<String> lockedGroupsReferences = new ArrayList<String>();
+                if (isGroupSubmit && !groups.isEmpty()) {
+                    for (Group group : groups) {
+                        String groupAssignmentReference = group.getReference() + "/assignment/" + a.getId();
+
+                        M_log.debug("Getting groups from reference: {}", groupAssignmentReference);
+                        lockedGroupsReferences.add(group.getReference());
+                        M_log.debug("Adding group: " +group.getReference());
+
+                        if (!aOldGroups.contains(group.getReference()) || !group.isLocked(groupAssignmentReference)) {     
+                            M_log.debug("locking group: "+group.getReference());
+                            group.lockGroup(groupAssignmentReference);
+                            M_log.debug("locked group: "+group.getReference());
+
+                            try {
+                                siteService.save(group.getContainingSite());
+                            }
+                            catch (IdUnusedException e)
+                            {
+                                M_log.warn(this + ":doUpdate_options  Cannot find site with id " + siteId);
```

Comment:
Do you really need "this" in the log statement?

M_log.warn(":doUpdate_options  Cannot find site with id {}", siteId);